site-src/geps/gep-1324.md

```diff
@@ -0,0 +1,127 @@
+# GEP-1324: Service Mesh in Gateway API
+
+* Issue: [#1324](https://github.com/kubernetes-sigs/gateway-api/issues/1324)
+* Status: Provisional
+
+## Overview
+
+Gateway API represents the next generation of traffic routing APIs in Kubernetes. While most of the current work for Gateway API is focused on the ingress use-case, support for service mesh implementations within the spec would be advantageous to the greater community. Therefore, this GEP serves as the genesis for developing common patterns for using Gateway API for east/west traffic within a mesh implementation.
+
+## Goals
+
+* Augment the Gateway API specification with the necessary models, resources, and modifications so that the xRoute primitives are useable by service mesh implementations
+* Ensure common policies applicable to both north/south and east/west are respected
+* Make sure that Gateway API ingress implementations can coexist and interoperate with mesh implementations
+* Remain agnostic as to the topology of mesh implementations’ data planes (i.e sidecar vs. non-sidecar)
+* Enable mesh implementations to have vendor-specific extensions for their own policies, etc, in line with existing extensibility offered by Gateway API
+* Cooperate with the greater Gateway API community on shared patterns/policies that are applicable for both use-cases (e.g. authorization policy)
+* Maintain the integrity of the namespace boundary for traffic routing and policy, making explicit those instances where users define exceptions
+
+## Non-Goals
+
+* Represent every service mesh feature within Gateway API
+* Duplicate concerns for the sake of “mesh-specific” resources
+* Overload existing Gateway API mechanisms when a new mechanism is more appropriate
+
+## Deferred Goals
+
+These are goals that we aren’t explicitly excluding, but will reconsider at a later point in time
+
+* Service mesh activation/enrollment (e.g. sidecar injection, namespace enrollment)
+* Flesh out egress gateway story
+  * How traffic gets to the gateway may start out as vendor specific, but perhaps Gateway API should standardize on exposing Routes for the egress gateway?
```

Comment:
Small suggestion about prominence and language: consider moving this to the bottom of the GEP under "Alternatives Considered" and use language that clearly indicates we mean not to cover them in _this_ GEP, but instead of a future GEP may introduce them.

Reply:
Are these really "alternatives" though? It feels more like scope we're shedding or deferring until later, hence the term "deferred goals". I do agree with moving it to the bottom though

Reply:
I can compromise on the `alternatives` bit, the bottom seems good.